### PR TITLE
config: Create tcctl vault secrets path delete command

### PR DESCRIPTION
**Pull request by odebretseni**

### CODE REVIEW

**Review by vikfedkiv — APPROVED** [review]

File: tc/app/vault/secrets/path/delete [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    shift 2
+  else
+    set "${__key}"
+    shift
+  fi
+done
+
+function _secret_delete() {
+  _check_bin_file vault
+  VAULT_TOKEN=$(tcctl vault token)
```
Comment (BrunIF) [inline tc/app/vault/secrets/path/delete:25]: Call the function here.

**Review by BrunIF — CHANGES_REQUESTED** [review]

File: tc/app/vault/file/upload [hunk 2]
```diff
@@ -26,7 +26,7 @@ done
 
 function _vault_upload() {
   _check_bin_file vault
-  VAULT_TOKEN=$(tcctl vault token)
+  VAULT_TOKEN=$(_vault_get_token)
```
Comment (BrunIF) [inline tc/app/vault/file/upload:29]: you can add here token or username with a password.

If you pass token - you can do everything with token 
If you pass user/password - then with them. 

And now you don't need use a file.

**Review by BrunIF — CHANGES_REQUESTED** [review]

File: tc/app/vault/file/upload (hunk 2 above)
Comment (odebretseni) [reply tc/app/vault/file/upload:29]: I do not understand why? all this is done by the function

File: tc/app/vault/functions [hunk 3]
```diff
@@ -14,13 +18,26 @@ export VAULT_ADDR
...
-  __MSG+="$__output"
   if [ "$_error_code" -eq "0" ]; then
     echo "true"
   else
     echo "false"
   fi
 }
+
+function _vault_get_token()  {
+  _check_bin_file vault
```
Comment (BrunIF) [inline tc/app/vault/functions:31]: _vault_get_token token {token}
_vault_get_token userpass {username} {password}

```bash
_type=$1
if [[ _type = "token" ]]; then
  _token=$2
fi
if [[ type = "userpass" ]]; then
  _username=$2
  _password=$3
fi

......

```

**Review by BrunIF — CHANGES_REQUESTED** [review]

File: tc/app/vault/file/download.help [hunk 4]
```diff
@@ -1,4 +1,37 @@
 
-Download the file from vault server
+tcctl vault file download --secret-path "PATH" --file "FILE"
 
-    tcctl vault file download --secret-path "path" --file "filename"
+tcctl vault file download --auth-type token \
+                          --token "VAULT_TOKEN" \
+                          --addr "VAULT_ADDR" \
+                          --secret-path "PATH" \
+                          --file "FILE"
```
Comment (BrunIF) [inline tc/app/vault/file/download.help:8]: Compact usage
```bash
tcctl vault file download \
  --auth-type token \
  --token "VAULT_TOKEN" \
  --addr "VAULT_ADDR" \
  --secret-path "PATH" \
  --file "FILE"
```

File: tc/app/vault/file/upload.help [hunk 5]
```diff
@@ -1,4 +1,37 @@
...
 
-    tcctl vault file upload --secret-path "path" --file filename
+tcctl vault file upload --auth-type token \
+                        --token "VAULT_TOKEN" \
+                        --addr "VAULT_ADDR" \
+                        --secret-path "PATH" \
+                        --file "FILE"
+
+tcctl vault file upload --auth-type userpass \
+                        --name "VAULT_NAME" \
```
Comment (BrunIF) [inline tc/app/vault/file/upload.help:11]: VAULT_NAME -> vault user name or vault login, etc
Only about key description.

File: tc/app/vault/functions [hunk 6]
```diff
@@ -1,26 +1,70 @@
...
     echo "false"
   fi
 }
+
+function _vault_authorization()  {
+  _check_bin_file vault
+  __auth_type=$1
+  if [[ "$__auth_type" = "userpass" ]]; then
+    __vault_name=$2
+    __vault_password=$3
```
Comment (BrunIF) [inline tc/app/vault/functions:32]: It is better to assign values ​​outside this condition

File: tc/app/vault/functions [hunk 7]
```diff
@@ -1,26 +1,70 @@
...
+    export VAULT_ADDR=$5
```
Comment (BrunIF) [inline tc/app/vault/functions:33]: `export` needless here. Please check.

File: tc/app/vault/functions [hunk 8]
```diff
@@ -1,26 +1,70 @@
...
+    _error_code=$?
+    if [ "$_error_code" -eq "0" ]; then
+      VAULT_TOKEN=$(echo "$__output" | sed 1,6d | tr -s ' ' | cut -f 2 -d ' ' | sed -n 1p)
+      export VAULT_TOKEN
+    else
+      __MSG="Error while creating vault token"
+      _message "$__MSG" "error" "$_error_code"
+    fi
+  elif [[ "$__auth_type" = "token" ]]; then
+    __vault_token=$4
```
Comment (BrunIF) [inline tc/app/vault/functions:44]: `VAULT_TOKEN` `__vault_token` you use two variables for one value. 
Do everything in one style.

**Review by BrunIF — CHANGES_REQUESTED** [review]

File: tc/app/vault/file/upload.help (hunk 5 above)
Comment (odebretseni) [reply tc/app/vault/file/upload.help:11]: we agreed to write a variable here, their description is below

`
--token                 "token from vault server"
--name                  "users name on the vault server"
--pass                  "users password on the vault server"
--addr                  "the vault server address"
`

File: tc/app/vault/functions (hunk 7 above)
Comment (odebretseni) [reply tc/app/vault/functions:33]: doesn't work without export, the environment doesn't see the variable

**Review by BrunIF — APPROVED** [review]

**Pull request merged**